src/Shared/classes/PathTranslator.ts

```diff
@@ -35,13 +37,24 @@ export class PathTranslator {
 		return (pathInfo: PathInfo) => path.join(to, path.relative(from, pathInfo.join()));
 	}
 
+	private checkFilenameWarnings(filePath: string) {
+		const baseName = path.basename(filePath);
+		const nameWarning = FILENAME_WARNINGS.get(baseName);
+		if (nameWarning && !this.warnedFiles.has(filePath)) {
+			warn(`PathTranslator: Dangerous file name! Change ${baseName} to ${nameWarning}! Full path: ${filePath}`);
```

Comment:
`!this.warnedFiles.has(filePath))` feels like a hack to get around the fact that this might run multiple times for the same file path. Instead, maybe we should just put this in `compileFiles.ts`?

I'd also like to move away from using the `warn()` function and instead use warning diagnostics defined in `diagnostics.ts`. If we move this into `compileFiles.ts`, that becomes a lot easier.

`warn` is only currently used in `RojoResolver.ts` and should probably eventually be removed altogether.

Also "Dangerous file name!" language is a little bit weird. Maybe instead:
`"Incorrect root file name! Change init.ts to index.ts. Full path: ..."`